jaeger/cmd/check.go

```diff
@@ -132,6 +140,53 @@ func jaegerCategory(hc *healthcheck.HealthChecker) *healthcheck.Category {
 	return healthcheck.NewCategory(linkerdJaegerExtensionCheck, checkers, true)
 }
 
+// JaegerDataPlaneCategory returns a healthcheck.Category containing checkers
+// to verify the jaeger injection
+func JaegerDataPlaneCategory(hc *healthcheck.HealthChecker) *healthcheck.Category {
+	return healthcheck.NewCategory(linkerdJaegerExtensionDataPlaneCheck, []healthcheck.Checker{
+		*healthcheck.NewChecker("data plane namespace exists").
+			WithHintAnchor("l5d-data-plane-exists").
+			Fatal().
+			WithCheck(func(ctx context.Context) error {
+				if hc.DataPlaneNamespace == "" {
+					// when checking proxies in all namespaces, this check is a no-op
+					return nil
+				}
+				return hc.CheckNamespace(ctx, hc.DataPlaneNamespace, true)
+			}),
+		*healthcheck.NewChecker("data-plane pods have tracing enabled").
+			WithHintAnchor("l5d-jaeger-data-plane-trace").
+			Warning().
+			WithCheck(func(ctx context.Context) error {
+				pods, err := hc.KubeAPIClient().GetPodsByNamespace(ctx, hc.DataPlaneNamespace)
+				if err != nil {
+					return err
+				}
+
+				return checkForTraceConfiguration(pods)
+			}),
+	}, true)
+}
+
+// checkForTraceConfiguration checks if the trace annotation is present
+// only for the pods with tracing enabled
+func checkForTraceConfiguration(pods []corev1.Pod) error {
+	var podsWithoutTraceConfig []string
+	for i := range pods {
+		pod := pods[i]
+		// Check for jaeger-injector annotation
+		if !labels.IsTracingEnabled(&pod) {
```

Comment:
Should we check here that tracing is not disabled at the namespace level, analogously as we do in viz check with tap?

Reply:
No, because there isn't a way to actually disable tracing and its enabled once the jaeger-injector is installed. Though the trace values can be overwritten using anotations.